tests/test_match.c

```diff
@@ -0,0 +1,76 @@
+#include <criterion/criterion.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include "match.h"
+
+/* Checks that new_match works */
+Test(match, new)
```

Comment:
These tests need to be fully implemented and passing before I will accept this PR.